bundles/org.openhab.core.model.sitemap/src/org/openhab/core/model/sitemap/fileconverter/AbstractSitemapFileGenerator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.model.sitemap.fileconverter;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.model.sitemap.sitemap.Sitemap;
+import org.osgi.service.component.annotations.Activate;
+
+/**
+ * {@link AbstractSitemapFileGenerator} is the base class for any {@link Sitemap} file generator.
+ *
+ * @author Mark Herwege - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractSitemapFileGenerator implements SitemapFileGenerator {
```

Comment:
This class should be moved into org.openhab.core.ui so that it can be used by YAML without creating a dependency to the DSL stuff.

Reply:
org.openhab.core.ui actually depends on org.core.model.core and core.model.sitemap. So as it is now, it doesn’t change much. In principle I do agree to move them. But it should be part of an effort to create a sitemap registry that does no depend on model.sitemap.

Reply:
Maybe it should be put in org.openhab.core ? I did not check if we have already something about sitemap in that bundle.

Reply:
I went ahead and created a sitemap registry: https://github.com/openhab/openhab-core/pull/5004